db/model.js

```diff
@@ -9,29 +9,46 @@ const retrieveTopics = () => {
     return db.query(sqlQuery).then((results) => results.rows)
 }
 
-const retrieveArticles = (topic) => {
-    
-    const queryValues = []
+const retrieveArticles = (topic, sort_by = 'created_at', order = 'desc') => {
 
-    let sqlQuery = `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, CAST(COUNT(comments.article_id) AS INTEGER) AS "comment_count"
-                        FROM articles
-                        LEFT JOIN comments ON articles.article_id = comments.article_id
-                        `
-
-    if(topic) {
-        console.log(topic)
-        sqlQuery += ` WHERE articles.topic = $1`;
-        queryValues.push(topic);
+    if(!['author', 'title', 'article_id', 'topic', 'created_at', 'votes', 'article_img_url'].includes(sort_by)) {
+        return Promise.reject({status:400, msg: 'Invalid column name to sort by'})
     }
 
-    sqlQuery += ` GROUP BY articles.article_id`
+    if(!['asc', 'desc', 'ASC', 'DESC'].includes(order)) {
+        return Promise.reject({status:400, msg: 'Invalid ordering request'})
+    }
+    
```

Comment:
instead of listing out the arrays in an if statement, could you make a green list earlier?